app/controllers/api/red_hat_cloud_service_providers_controller.rb

```diff
@@ -0,0 +1,40 @@
+module Api
+  class RedHatCloudServiceProvidersController < BaseController
+    def red_hat_cloud_service_providers_search_conditions
+      { :type => provider_types }
+    end
+
+    def find_red_hat_cloud_service_providers(id)
```

Comment:
Should this method be singular since it only finds a single provider?

Reply:
not unless we change the base api, method it looks for is find_<collection_endpoint>